src/librustc_mir/interpret/intrinsics.rs

```diff
@@ -46,8 +46,11 @@ crate fn eval_nullary_intrinsic<'tcx>(
     def_id: DefId,
     substs: SubstsRef<'tcx>,
 ) -> InterpResult<'tcx, &'tcx ty::Const<'tcx>> {
-    let tp_ty = substs.type_at(0);
     let name = &*tcx.item_name(def_id).as_str();
+    if name == "is_const_eval" {
+        return Ok(ty::Const::from_bool(tcx, true));
```

Comment:
Won't this mean that it returns `true` in Miri, which seems odd (for now, anyway)?

This should query a machine flag to determine the intrinsic's return value.

Reply:
> Won't this mean that it returns true in Miri, which seems odd (for now, anyway)?

This returns `true` in miri. Were it to return `false`, `miri` would fail trying to do something it doesn't support.

Reply:
I don't think this is odd at all given that miri already provides `cfg!(miri)` which can be used to the same effect. 

> This should query a machine flag to determine the intrinsic's return value.

Does `cfg!(miri)` do this?

Reply:
The difference is that miri is more like codegen and not like const eval in its semantics. Cfgs flags are their own world independent of the runtime

Reply:
Miri is implementing Rust's run-time semantics as much as it can. So `is_const_eval` must return `false`.

If you actually want to check "is this a restricted interpreted environment that does not support SIMD", then you should pick an appropriate name for the intrinsic.

Reply:
> Miri is implementing Rust's run-time semantics as much as it can. So is_const_eval must return false.

I could `cfg(miri)` `is_const_eval` to always return false, but then miri won't be able to execute any code for which `is_const_eval` makes sense using.